banzai/dbs.py

```diff
@@ -292,25 +297,41 @@ class TelescopeMissingException(Exception):
     pass
 
 
-def get_telescope_id(site, instrument, db_address=_DEFAULT_DB):
-    # TODO:  This dies if the telescope is not in the telescopes table. Maybe ping the configdb?
+def get_telescope_id(header, db_address=_DEFAULT_DB):
     db_session = get_session(db_address=db_address)
     criteria = (Telescope.site == site) & (Telescope.instrument == instrument)
     telescope = db_session.query(Telescope).filter(criteria).first()
     db_session.close()
     if telescope is None:
         err_msg = '{site}/{instrument} is not in the database.'.format(site=site,
                                                                        instrument=instrument)
-        raise TelescopeMissingException(err_msg)
-    return telescope.id
+        logger.error(err_msg, extra={'tags': {'site': site, 'instrument': instrument}})
+        telescope_id = None
+    else:
+        telescope_id = telescope.id
+    return telescope_id
 
 
-def get_telescope(telescope_id, db_address=_DEFAULT_DB):
+def query_for_telescope(db_address, site, instrument):
+    # Short circuit
+    if site is None or instrument is None:
+        return None
     db_session = get_session(db_address=db_address)
-    telescope = db_session.query(Telescope).filter(Telescope.id == telescope_id).first()
+    criteria = (Telescope.site == site) & (Telescope.instrument == instrument)
+    telescope = db_session.query(Telescope).filter(criteria).first()
     db_session.close()
+    return telescope
+
+
+def get_telescope(header, db_address=_DEFAULT_DB):
+    site = header.get('SITEID')
+    instrument = header.get('INSTRUME')
+    telescope = query_for_telescope(db_address, site, instrument)
+    if telescope is None:
+        telescope = query_for_telescope(db_address, site, header.get('TELESCOPE'))
     if telescope is None:
-        raise TelescopeMissingException('Telescope ID {id} is not in the database.'.format(id=telescope_id))
+        err_msg = '{site}/{instrument} is not in the database.'.format(site=site, instrument=instrument)
+        logger.error(err_msg, extra={'tags': {'site': site, 'instrument': instrument}})
     return telescope
 
```

Comment:
What if you raise a `TelescopeMissingException` in `query_for_telescope`, and then use try/except in `get_telescope` instead. I guess the downside is that the error will pop up every time you give it an NRES frame. But something like this:

```
def query_for_telescope(db_address, site, instrument):
    db_session = get_session(db_address=db_address)
    criteria = (Telescope.site == site) & (Telescope.instrument == instrument)
    telescope = db_session.query(Telescope).filter(criteria).first()
    db_session.close()
    if telescope is None:
        err_msg = '{site}/{instrument} is not in the database.'.format(site=site, instrument=instrument)
        raise TelescopeMissingException(err_msg)
    return telescope


def get_telescope(header, db_address=_DEFAULT_DB):
    site = header.get('SITEID')
    instrument = header.get('INSTRUME')
    try:
        telescope = query_for_telescope(db_address, site, instrument)
    except TelescopeMissingException:
        telescope = query_for_telescope(db_address, site, header.get('TELESCOPE'))
    except TelescopeMissingException:
        telescope = None
    return telescope

```

Reply:
I like this

Reply:
Although we still want to set self.instrument = header.get('TELESCOP')
for the image object after searching the telescope, so that the designations are clear down the line

Reply:
IS there some way we could get it to not throw an error for NRES frames every time?